src/topology/quasi_separated.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.subset_properties
+import topology.sets.compacts
+
+/-!
+# Quasi-separated spaces
+
+A topological space is quasi-separated if the intersections of any pairs of compact open subsets
+are still compact.
+Notible examples include spectral spaces and Hausdorff spaces.
+
+## Main results
+
+- `is_quasi_separated`: A subset `s` of a topological space is quasi-separated if the intersections
+of any pairs of compact open subsets of `s` are still compact.
+- `quasi_separated_space`: A topological space is quasi-separated if the intersections of any pairs
+of compact open subsets are still compact.
+- `quasi_separated_space.of_open_embedding`: If `f : α → β` is an open embedding, and `β` is
+  a quasi-separated space, then so is `α`.
+
+-/
+
+variables {α β : Type*} [topological_space α] [topological_space β] {f : α → β}
+
+/-- A subset `s` of a topological space is quasi-separated if the intersections of any pairs of
+compact open subsets of `s` are still compact.
+
+Note that this is equiavlent to `s` being a `quasi_separated_space` only when `s` is open. -/
+def is_quasi_separated (s : set α) : Prop :=
+∀ (U V : set α), U ⊆ s → is_open U → is_compact U → V ⊆ s →
+  is_open V → is_compact V → is_compact (U ∩ V)
+
+/-- A topological space is quasi-separated if the intersections of any pairs of compact open
+subsets are still compact. -/
+@[mk_iff]
+class quasi_separated_space (α : Type*) [topological_space α] : Prop :=
+(inter_is_compact : ∀ (U V : set α),
+  is_open U → is_compact U → is_open V → is_compact V → is_compact (U ∩ V))
+
+lemma is_quasi_separated_univ (α : Type*) [topological_space α] :
+  is_quasi_separated (set.univ : set α) ↔ quasi_separated_space α :=
+begin
+  rw quasi_separated_space_iff,
+  simp [is_quasi_separated],
+end
+
+lemma is_quasi_separated.image_of_embedding (h : embedding f) (s : set α)
+  (H : is_quasi_separated s) : is_quasi_separated (f '' s) :=
```

Comment:
I think that dot notation will work better if you assume `{s} (hs : is_quasi_separated s) (hf : embedding f)` in this order. E.g., this way you won't get strange error messages if you write `hs.image_of_embedding`.

Reply:
Really, `embedding` should be ported to new homs.

Reply:
`embedding` is a predicate, not a bundled morphism.